### PR TITLE
Adds ActiveFedora::Checksum class to encapsulate file digest info.

### PR DESCRIPTION
**Pull request by dchandekstark**
Adds `#checksum` method to ActiveFedora::File.

### CODE REVIEW

**Comment by jcoyne** [comment]
:+1: @awead care to have a look?

File: lib/active_fedora/file.rb [hunk 1]
```diff
@@ -134,6 +134,10 @@ def metadata
       @metadata ||= ActiveFedora::WithMetadata::MetadataNode.new(self)
     end
 
+    def checksum
+      ActiveFedora::Checksum.new(self)
```
Comment (awead) [inline lib/active_fedora/file.rb]: Maybe:
```
@checksum ||= ActiveFedora::Checksum.new(self)
```
and then set `@checksum = nil` under #refresh.

Comment (dchandekstark) [inline lib/active_fedora/file.rb]: Personally I don't think the operation cost warrants the complexity of caching.  Or if it does, perhaps it belongs on `ActiveFedora::File::Attributes#digest`.  @jcoyne

Comment (awead) [inline lib/active_fedora/file.rb]: Yeah, I'd buy that.

Comment (dchandekstark) [inline lib/active_fedora/file.rb]: OK, well maybe we merge this and address any caching separately if anybody thinks that's important?

**Pull request merged**